releases/v23_0_0.rst

```diff
@@ -0,0 +1,270 @@
+| Git tag                                   | 23.0.0     |
++-------------------------------------------+------------+
+| :doc:`EUPS distrib </install/newinstall>` | v23\_0\_0  |
++-------------------------------------------+------------+
+
+
+This release is based on the ``w_2021_40`` weekly build, with 29 tickets backported to support the Data Preview 0.2 data release.
+
+The notes below highlight significant technical changes to the Science Pipelines codebase in this release.
+For a complete list of changes made, see :doc:`tickets/v23_0_0`.
+
+The `Characterization Metric Report (DMTR-351) <https://dmtr-351.lsst.io/>`_ describes the scientific performance of this release in terms of scientific performance metrics.
+
+If you have questions or comments about this release, visit our `community forum <https://community.lsst.org>`_ for advice.
+
+- :ref:`release-v23-0-0-functionality`
+- :ref:`release-v23-0-0-interface`
+- :ref:`release-v23-0-0-pending-deprecations`
+- :ref:`release-v23-0-0-deprecations`
+
+*See also:*
+
+- :doc:`Installation instructions </install/index>`
+- :doc:`Known issues </known-issues>`
+- `Doxygen Documentation`__
+
+__ http://doxygen.lsst.codes/stack/doxygen/xlink_master_2021_09_30_20.02.36/
+
+
+.. _release-v23-0-0-functionality:
+
+Major New Features
+==================
+
+- :ref:`release-v23-0-0-gen3`
+- :ref:`release-v23-0-0-meas_extensions_gaap`
+- :ref:`release-v23-0-0-trailed`
+- :ref:`release-v23-0-0-cp_verify`
+- :ref:`release-v23-0-0-ci_imsim`
+
+.. _release-v23-0-0-gen3:
+
+Generation 3 Middleware
+-----------------------
+Generation 3 Middleware is being used in production for the DP0.2 data release. This release is that last with functional generation 2 middleware. See :ref:`release-v23-0-0-deprecate-gen2`. Some highlights include:
+
+* Added a new facility for creating lightweight butlers that pre-fills a local SQLite registry. This "execution butler" can allow a pipeline to be executed without talking to the main registry, reducing contention for connections to shared registries. :jirab:`DM-28646`
+* Log diagnostic information when QuantumGraphs are empty because the initial query yielded no results. At present, these diagnostics only cover missing input datasets, which is a common way to get an empty QuantumGraph, but not the only way. :jirab:`DM-31583`
+* Add ability to cache datasets locally when using a remote file store. This can significantly improve performance when retrieving components from a dataset. :jirab:`DM-13365`
+
+For more information, see detailed notes on changes to the following middleware packages on this release: `daf_butler <https://pipelines.lsst.io/v/daily/modules/lsst.daf.butler/CHANGES.html>`_, `pipe_base <https://pipelines.lsst.io/v/daily/modules/lsst.pipe.base/CHANGES.html>`_, `ctrl_mpexec <https://pipelines.lsst.io/v/daily/modules/lsst.ctrl.mpexec/CHANGES.html>`_, and `ctrl_bps <https://pipelines.lsst.io/v/daily/modules/lsst.ctrl.bps/CHANGES.html>`_.
```

Comment:
These are external links to the daily docs, which is a no-go. These should work as internal links:

```suggestion
For more information, see detailed notes on changes to the following middleware packages on this release:

- :doc:`lsst.daf.butler </modules/lsst.daf.base/CHANGES>`_,
- :doc:`lsst.pipe.base </modules/lsst.pipe.base/CHANGES>`_,
- :doc:`lsst.ctrl.mpexec </modules/lsst.ctrl.mpexec/CHANGES>`_,
- :doc:`lsst.ctrl.bps </modules/lsst.ctrl.bps/CHANGES>`_.
```

Reply:
I have to undo this change.

The 23.0.0 middleware package release notes aren't in 23.0.0 (because they were compiled after the final release candidate rc5), so those internal links don't exist until we build 23.0.1.  e.g.: https://lsst.ncsa.illinois.edu/~yusra/23.0.0/modules/lsst.ctrl.mpexec/index.html#changes

https://lsstc.slack.com/archives/C2JPMCF5X/p1639601968158000?thread_ts=1639601844.157900&cid=C2JPMCF5X

Recommendations are welcome. I could just not link anything and say that more details are on the way? I could leave the dead links which will magically get switched on for v23.0.1.

Reply:
Forgot to mention: I'll use internal links on the version that gets merged to main.

OK how about this for v23.0.x: Added a TODO DM-33028 for the internal links to be backported. Made the external links go to github, so that it'll be super obvious that they're not looking at the 23.0.0 doc build anymore...

Reply:
Yeah, pragmatism for now and patching it is fine if that's what you want. As you can imagine, we can't do this style of linking in general because these v23 release notes will someday be linking to the v28 release notes for butler :)

We definitely need to formalize the change log management of middleware and put it into the Science Pipelines release procedure; I'm not thrilled with how the middleware team kind of forked the release notes process unilaterally.

Reply:
Middleware did not do anything that required any changes to process so I'm not sure why this is causing upset. I wanted to keep track of changes as we made them and we adopted a scheme that made that easy. The global release notes were not affected. I made the middleware release notes for v23 after rc5 so they don't appear in the butler docs. If there was an rc6 then they would appear but that doesn't affect anything here. If you link generically to CHANGES then there are sections for each release.

Reply:
> We definitely need to formalize the change log management of middleware and put it into the Science Pipelines release procedure; 

We prob should have done one more rc.  This is the first time we've had a long-lived release branch tied to a data release, with more than a couple backported tickets and release candidates. So, we're still working out the final procedure in this new realm. 

We were going to try using towncrier in more packages next cycle. It was kind of nice to select some highlights to copy and paste.